ls1intum/Artemis
# [Enhancement] Feedback Alignment

### Checklist

### Motivation and Context


This PR removes the general feedback option and aligns the validation of feedback of all exercise types. Referenced feedback will only need a points, whereas unreferenced will need text and points. 

### Description

- Adaption of the database to transform all general feedback (Type: Manual or Null, Reference: Null) to unreferenced feedback (Type: Manual_Unreferenced, Reference: null)
- Remove all general feedback on the client side
- Align feedback validation to be equal for all exercise types
### Steps for Testing

!!!IMPORTANT!!!
First test it locally, and see if all database changes are done correctly (see also screenshots below)

Test for database change and removed general feedback
1. Log in to Artemis
2. Navigate to Course Administration
3. Before testing this PR make sure that you have exercises (file, text, programming, modeling) that have assessment with a general feedback. You can create a new exercise while another PR or develop is deployed on the ts1 (when this PR is deployed there will not be a possibility to add general feedback anymore)
4. Open the already assessed submission with general feedback and check if it is now displayed as unreferenced feedback (see screenshots below)
5. Check also the student view if everything is displayed accordingly (see screenshots below)

Test for validation:
1. Create a new exercise (programming, modeling, text and file upload) and participate
2. Assess the exercise and check that for unreferenced you always need to set the points and text and for referenced you just need to set the points.
3. Try different scenarios (see screenshots below)
4. Check also the student view if everything is displayed accordingly (see screenshots below)

### Test Coverage



Nothing changed.

### Screenshots



Before: Assessment with general feedback:
<img width="1140" alt="image" src="https://user-images.githubusercontent.com/41108487/108458633-5358ca00-7275-11eb-829b-42b41bcdaa7a.png">

database:
<img width="988" alt="image" src="https://user-images.githubusercontent.com/41108487/108458701-6ec3d500-7275-11eb-82cd-bd21c241b649.png">



After: General feedback is transformed to unreferenced feedback:
<img width="1229" alt="image" src="https://user-images.githubusercontent.com/41108487/108459352-c878cf00-7276-11eb-856b-0bb314a298e8.png">

database:
<img width="981" alt="image" src="https://user-images.githubusercontent.com/41108487/108459392-d75f8180-7276-11eb-8f13-8198e1f8c32e.png">


Unreferenced Feedback validation must have points and text:
<img width="1671" alt="image" src="https://user-images.githubusercontent.com/41108487/108607918-b6f10d80-73c3-11eb-98d5-5693fe282641.png">
<img width="1667" alt="image" src="https://user-images.githubusercontent.com/41108487/108607934-c53f2980-73c3-11eb-8039-787a8372ffaf.png">

Referenced Feedback must only have points:
<img width="1656" alt="image" src="https://user-images.githubusercontent.com/41108487/108607960-ef90e700-73c3-11eb-8204-8a3f128466bb.png">

If you have both referenced and unreferenced you must also fulfil the validation:
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/41108487/108608000-27982a00-73c4-11eb-998b-0e2fab0a5c17.png">
<img width="1656" alt="image" src="https://user-images.githubusercontent.com/41108487/108608007-31ba2880-73c4-11eb-96a2-8e557067e784.png">

Assessment of Text Exercise:
<img width="1428" alt="image" src="https://user-images.githubusercontent.com/41108487/109122473-f4012b00-7748-11eb-9a16-1197492cfa67.png">
Student View of Assessment:
![image](https://user-images.githubusercontent.com/41108487/109122419-daf87a00-7748-11eb-8484-290227940034.png)
